src/site/content/en/blog/sign-up-form-best-practices/index.md

```diff
@@ -0,0 +1,408 @@
+
+```js
+// auth2 is initialized with gapi.auth2.init()
+if (auth2.isSignedIn.get()) {
+  var profile = auth2.currentUser.get().getBasicProfile();
+  console.log(`Email: ${profile.getEmail()}`);
+}
+
+```
+
+{: #obvious-account-details}
+
+Once a user has signed in, make it clear how to access account details. In particular, make it 
+obvious how to [change or reset passwords](#password-change). 
+
+
+## Cut form clutter {: #cut-clutter}
+
+In the sign-up flow, your job is to minimize complexity and keep the user focused. Cut the clutter. 
+This is not the time for distractions and temptations!
+
+<figure class="w-figure">
+   <video controls autoplay loop muted class="w-screenshot">
+     <source src="https://samdutton.com/avoid-distractions.mp4" type="video/mp4">
+   </video>
+  <figcaption class="w-figcaption">Don't distract users from completing sign-up.</figcaption>
+</figure>
+
+On sign-up, ask for as little as possible. Collect additional user data (such as name and address) 
+only when you need to, and when the user sees a clear benefit from providing that data. Bear in mind 
+that every item of data you communicate and store incurs cost and liability.
+
+Don't double up your inputs just to make sure users get their contact details right. That slows down 
+form completion and doesn't make sense if form fields are autofilled. Instead, send a confirmation 
+code to the user once they've entered their contact details, then continue with account creation 
+once they respond. This is a common sign-up pattern: users are used to it.
+
+You may want to consider password-free sign-in by sending users a code every time they sign in on 
+a new device or browser. Sites such as Slack and Medium use a version of this. 
+
+<figure class="w-figure">
+   <video controls autoplay loop muted class="w-screenshot">
+     <source src="https://samdutton.com/medium-sign-in.mp4" type="video/mp4">
+   </video>
+  <figcaption class="w-figcaption">Password-free sign-in on medium.com.</figcaption>
+</figure>
+
+As with federated login, this has the added benefit that you don't have to manage user passwords. 
+
+## Consider session length {: #session-length}
+ 
+Whatever approach you take to user identity. you'll need to make a careful decision about session 
+length: how long the user remains logged in, and what might cause you to log them out.
+
+There's no one hard-and-fast rule, but you need to consider mobile versus desktop, and whether 
+users are sharing on desktop, or sharing devices. 
+
+{% Aside %}
+You can get around some of the issues of shared devices by enforcing re-authentication for sensitive 
+features, for example when a purchase is made or an account updated. You can find out more about 
+ways to implement re-authentication from [Verify phone numbers on the web with the Web OTP API](/web-otp/#is-it-safe-to-use-sms-as-a-way-to-authenticate) and the codelab [Your First WebAuthn App](https://codelabs.developers.google.com/codelabs/webauthn-reauth/#0).
```

Comment:
Web OTP is not about re-authentication. Please remove the link. (SMS OTP form best practices I'm working on neither. Just in case.)

Reply:
Removed. (I meant that readers should look at your comments in that article, but it sounds like a recommendation to use Web OTP for re-authentication.)